code/modules/mob/living/living.dm

```diff
@@ -123,22 +123,26 @@ default behaviour is:
 		now_pushing = 0
 		spawn(0)
 			..()
-			if (!istype(AM, /atom/movable))
+			if (!istype(AM, /atom/movable) || AM.anchored)
+				if(confused && prob(50) && m_intent=="run")
+					Paralyse(1)
+					playsound(loc, "punch", 25, 1, -1)
+					visible_message("<span class='warning'>[src] [pick("ran", "slammed", "waltzed")] into \the [AM]!</span>")
```

Comment:
A waltz is a dance, it's a pretty odd term to describe someone running headlong into an obstacle.